src/wallet/rpcwallet.cpp

```diff
@@ -2914,9 +2888,8 @@ static UniValue listunspent(const JSONRPCRequest& request)
         cctl.m_avoid_address_reuse = false;
         cctl.m_min_depth = nMinDepth;
         cctl.m_max_depth = nMaxDepth;
-        auto locked_chain = pwallet->chain().lock();
         LOCK(pwallet->cs_wallet);
```

Comment:
Move this up and remove the lock in L2922.

Reply:
You mean the main wallet lock just after ? I think some other wallet locks can also be removed in other places and that would be better to do it in a follow-up PR with dedicated thinking.